packages/runtime/container-runtime/src/dataStoreContexts.ts

```diff
@@ -81,9 +81,23 @@ export class DataStoreContexts implements Iterable<[string, FluidDataStoreContex
 	public delete(id: string): boolean {
 		this.deferredContexts.delete(id);
 		this.notBoundContexts.delete(id);
+
+		// Stash the context here in case it's requested in this session, we can log some details about it
+		const context = this._contexts.get(id);
+		this._recentlyDeletedContexts.set(id, context);
+
 		return this._contexts.delete(id);
 	}
 
+	private readonly _recentlyDeletedContexts: Map<string, FluidDataStoreContext | undefined> =
+		new Map();
+
+	public getRecentlyDeletedContextPath(id: string): string | undefined {
+		// context.packagePath will throw if pkg is undefined - just reach in and get the value
+		const context = this.get(id) as { pkg: readonly string[] | undefined } | undefined;
+		return context?.pkg?.join("/");
```

Comment:
As you mentioned in the PR description, this will only be present in cases where data store has already been loaded.
We can get the package here by calling `await context.getInitialSnapshotDetails()` (we may have to log async). But, I think this may pose problems with the new data virtualization feature where we will have to download the attributes blob from the server.

Reply:
Not really sure what the best course here is.  I'd rather log immediately, maybe we have a second opportunistic log that fires after we await the snapshot?  I'll try it.  And, not sure if the problem with data virtualization is present in the current code or would be a problem for an upcoming change, can you elaborate?

Reply:
I added async logging (kind of ugly), take a look.